testsuites/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/manifest/SPARQL10ManifestTest.java

```diff
@@ -0,0 +1,183 @@
+
+	static final Logger logger = LoggerFactory.getLogger(SPARQL10ManifestTest.class);
+
+	private static final boolean REMOTE = false;
+
+	public static TestSuite suite(SPARQLQueryTest.Factory factory)
+		throws Exception
+	{
+		final String manifestFile;
+		final File tmpDir;
+
+		if (REMOTE) {
+			manifestFile = "http://www.w3.org/2001/sw/DataAccess/tests/data-r2/manifest-evaluation.ttl";
+			tmpDir = null;
+		}
+		else {
+			URL url = SPARQL10ManifestTest.class.getResource("/testcases-sparql-1.0-w3c/data-r2/manifest-evaluation.ttl");
+
+			if ("jar".equals(url.getProtocol())) {
+				// Extract manifest files to a temporary directory
+				try {
+					tmpDir = FileUtil.createTempDir("sparql-evaluation");
+
+					JarURLConnection con = (JarURLConnection)url.openConnection();
+					JarFile jar = con.getJarFile();
+
+					ZipUtil.extract(jar, tmpDir);
+
+					File localFile = new File(tmpDir, con.getEntryName());
+					manifestFile = localFile.toURI().toURL().toString();
+				}
+				catch (IOException e) {
+					throw new AssertionError(e);
+				}
+			}
+			else {
+				manifestFile = url.toString();
+				tmpDir = null;
+			}
+		}
+
+		TestSuite suite = new TestSuite(factory.getClass().getName()) {
+
+			@Override
+			public void run(TestResult result) {
+				try {
+					super.run(result);
+				}
+				finally {
+					if (tmpDir != null) {
+						try {
+							FileUtil.deleteDir(tmpDir);
+						}
+						catch (IOException e) {
+							System.err.println("Unable to clean up temporary directory '" + tmpDir + "': "
+									+ e.getMessage());
+						}
+					}
+				}
+			}
+		};
+
+		Repository manifestRep = new SailRepository(new MemoryStore());
+		manifestRep.initialize();
+		RepositoryConnection con = manifestRep.getConnection();
+
+		addTurtle(con, new URL(manifestFile), manifestFile);
+
+		String query = "SELECT DISTINCT manifestFile FROM {x} rdf:first {manifestFile} "
+				+ "USING NAMESPACE mf = <http://www.w3.org/2001/sw/DataAccess/tests/test-manifest#>, "
+				+ "  qt = <http://www.w3.org/2001/sw/DataAccess/tests/test-query#>";
+
+		TupleQueryResult manifestResults = con.prepareTupleQuery(QueryLanguage.SERQL, query, manifestFile).evaluate();
+
+		while (manifestResults.hasNext()) {
+			BindingSet bindingSet = manifestResults.next();
+			String subManifestFile = bindingSet.getValue("manifestFile").stringValue();
+			suite.addTest(SPARQLQueryTest.suite(subManifestFile, factory));
+		}
+
+		manifestResults.close();
+		con.close();
+		manifestRep.shutDown();
+
+		logger.info("Created aggregated test suite with " + suite.countTestCases() + " test cases.");
+		return suite;
+	}
+
+	static void addTurtle(RepositoryConnection con, URL url, String baseURI, Resource... contexts)
+		throws IOException, RepositoryException, RDFParseException, RDFHandlerException
+	{
+		if (baseURI == null) {
+			baseURI = url.toExternalForm();
+		}
+
+		InputStream in = url.openStream();
+
+		try {
+			OpenRDFUtil.verifyContextNotNull(contexts);
```

Comment:
This was probably here in the past, but generally is the first line in a method in other places in the codebase.